discovery-infra/virsh_cleanup.py

```diff
@@ -76,9 +78,10 @@ def clean_networks(skip_list, resource_filter):
 
 
 def clean_virsh_resources(skip_list, resource_filter):
-    clean_domains(skip_list, resource_filter)
-    clean_pools(skip_list, resource_filter)
-    clean_networks(skip_list, resource_filter)
+    with utils.file_lock_context():
```

Comment:
why we need lock here?

Reply:
because libvirt command keeps failing when executing them in parallel.